docs/src/main/asciidoc/kafka-schema-registry-avro.adoc

```diff
@@ -0,0 +1,604 @@
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.equalToIgnoringCase;
+
+@QuarkusTest
+// register the class that sets up Testcontainers:
+@QuarkusTestResource(KafkaAndSchemaRegistryTestResource.class)
+public class MovieResourceTest {
+
+    @TestHTTPResource("/consumed-movies")
+    URI consumedMovies;
+
+    @Test
+    public void testHelloEndpoint() {
+        // create a client for `ConsumedMovieResource` and collect the consumed resources in a list
+        Client client = ClientBuilder.newClient();
+        WebTarget target = client.target(consumedMovies);
+
+        List<String> received = new CopyOnWriteArrayList<>();
+
+        SseEventSource source = SseEventSource.target(target).build();
+        source.register(inboundSseEvent -> received.add(inboundSseEvent.readData()));
+
+        // in a separate thread, feed the `MovieResource`
+        ExecutorService movieSender = startSendingMovies();
+
+        source.open();
+
+        // check if, after at most 5 seconds, we have at last 2 items collected, and they are what we expect:
+        await().atMost(5000, MILLISECONDS).until(() -> received.size() >= 2);
+        assertThat(received, Matchers.hasItems("'The Shawshank Redemption' from 1994",
+                "'12 Angry Men' from 1957"));
+        source.close();
+
+        // shutdown the executor that is feeding the `MovieResource`
+        movieSender.shutdown();
+    }
+
+    private ExecutorService startSendingMovies() {
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
+        executorService
+                .execute(
+                        () -> {
+                            while (true) {
+                                given()
+                                        .contentType(ContentType.JSON)
+                                        .body("{\"title\":\"The Shawshank Redemption\",\"year\":1994}")
+                                    .when().post("/movies")
+                                    .then()
+                                        .statusCode(202);
+                                given()
+                                        .contentType(ContentType.JSON)
+                                        .body("{\"title\":\"12 Angry Men\",\"year\":1957}")
+                                    .when().post("/movies")
+                                    .then()
+                                        .statusCode(202);
+                                try {
+                                    Thread.sleep(200L);
+                                } catch (InterruptedException e) {
+                                    break;
+                                }
+                            }
+                        }
+                );
+        return executorService;
+    }
+
+}
+----
+
```

Comment:
Maybe add the native version? 
Avro should be supported in native.

Reply:
native version is already in the code generated by the quarkus plugin, I w ill mention it here and add info on how to execute it